### PR TITLE
Disable submit button when password fail validation

### PR DESCRIPTION
**Pull request by jingcheng16**
## Product Description

When web user accept invitation and trying to create account or when mobile user receive sms invitation and trying to confirm account: they're able to submit even their password failed password validation.
After changes, the button will be disabled until they pass the validation.

Before fix:
image
After fix:
image

## Technical Summary

Ticket: https://dimagi-dev.atlassian.net/browse/QA-4314

## Feature Flag

Enable two-stage user provisioning (users confirm and set their own passwords via sms).

### Safety story

This button is originally included in fieldset, whose class is "check-password".
I moved the button back into the fieldset, thus `satisfyPassword` will work.

### QA Plan

Ticket: https://dimagi-dev.atlassian.net/browse/QA-4314

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/users/templates/users/partials/register_user_form_fields.html
Comment (jingcheng16) [reply corehq/apps/users/templates/users/partials/register_user_form_fields.html:35]: `submitCheck` will also run if user tabs to the button and press enter.

**Comment by jingcheng16** [comment]
@orangejenny 
Hi Jenny, I find a bug after I create this PR, the button "Confirm Account" becomes "Create Account" in mobile worker confirmation dialog. (I didn't noticed when look at the code because the button label are so similar, but noticed now since I upload "before fix" and "after fix" to compare.)
The only solution I can think of is: move the whole fieldset back to the template, which will resulted in same code appeared twice: in [`accept_invite.html`](https://github.com/dimagi/commcare-hq/blob/3aeba163cd7eeffd8fb0e62a2ca8f71e4e3c95d9/corehq/apps/users/templates/users/accept_invite.html) as well as [`commcare_user_confirm_account.html`](https://github.com/dimagi/commcare-hq/blob/3aeba163cd7eeffd8fb0e62a2ca8f71e4e3c95d9/corehq/apps/users/templates/users/commcare_user_confirm_account.html). (But Cory extracted them out 2 years ago, I understand the reason why Cory extracted them out, and I want to keep the extraction)
I cannot think of any better solution. Do you have any idea?

**Review by orangejenny — APPROVED** [review]

**Comment by jingcheng16** [comment]
@orangejenny 
Hi Jenny, I go with the first option in the end. Thank you for the suggestions!

**Comment by orangejenny** [comment]
@jingcheng16 Cool, this looks good. I noticed you have the "awaiting QA" label on this PR, but since this is being merged into another PR instead of master, you can merge it. (It's up to you - sometimes it can become a hassle to keep a bunch of sub-branches on staging as they're being QAed.)

**Comment by jingcheng16** [comment]
@orangejenny 
Hi Jenny, I did have a question about: should I merge my `child branch` into my `parent branch` once child get approved, or should I wait until my `parent branch` is merged and then merge my `child branch` directly to `master`. I asked @joxl and he recommended the latter. I'm wondering do we have a document about this?

My concerns are:
-  My `parent branch` is still under review. I'm still having conversation with Matt. If I merge it into parent branch, will he get confused when he see new changes?
- In case I failed the QA test, maybe it's better for me to push my changes in this PR rather than in a parent PR?

I'm curious about in what cases will it become a hassle? One case I can think of is, I made changes in `parent branch`, which resulted in a conflict between `child branch` and `parent branch`

**Comment by joxl** [comment]
> since this is being merged into another PR instead of master, you can merge it.

@orangejenny, is this workflow common? I've always avoided doing this to prevent "already reviewed" commits piling on top of another open PR. TBH, I haven't used this workflow myself (merging a child branch into an open PR branch), maybe GitHub makes that action "convenient" on the destination branch (similar to how merging master into a branch doesn't become burdensome)?

In the case of this PR, my general feeling is that the parent branch will be merged soon enough to avoid inconvenience.

**Comment by orangejenny** [comment]
Interesting, maybe this is a case where I just assumed everyone does what I do (shame on me). It's not just me, though - Graham did this for his ERM work, and USH has done it for our big case search changes.

FWIW, these three branches do conflict - I just added `jc/minimum-pwd-length+jc/bulk-load-password-validation+jc/disable-submit-when-password-fail-validation` to staging this morning. (It was only one conflict, but rebuildstaging had said there were conflicts across all 3, so I combined them all, since rebuildstaging is slow and it's annoying to have it fail repeatedly.) I also have a bad habit of forgetting to add branches to staging.yml.

As a reviewer of the parent PR, I don't mind child PRs getting merged in even if I'm in the middle of a review, because I'm virtually always reviewing by commit, so I just skip the commits from child PRs. I have GitHub notifications set to email me on all pushes, so I get notified when a child PR is merged and it's clear which commits to skip. But our docs on this do recommend not merging child PRs until the parent PR is merged: https://github.com/dimagi/open-source/blob/master/docs/Writing_PRs.md#very-large-changes-that-cant-be-merged-in-smaller-chunks

As the developer, if there are more bugs in the child PR after it's merged, I just make more child PRs.

That was quite long-winded. Ultimately it's definitely Jing's and Joel's call, since you two are working closely together, I agree it doesn't much matter when the PRs aren't open for long.

**Comment by joxl** [comment]
> the parent branch will be merged soon enough to avoid inconvenience.

Ah, this statement was ignorant of the staging process. I haven't run into that problem in this scenario, so I may have some blind spots around this process in regards to staging deploys. I'll look into this to make sure I'm not overlooking any details or workflows.

That's for the detailed explanation @orangejenny, I appreciate the verbosity because it provides very useful context.

**Comment by jingcheng16** [comment]
@orangejenny 
Sorry, I noticed the conflicts when I replied to your comments yesterday, and I planed to fix it after I sent the reply. Then I forgot. 😂
But thank you for addressing my concerns, I now have another option when I have child branch on an open PR. Appreciate that. 🙏

**Pull request merged**
